qemu/tests/remote_server_disconnected.py

```diff
@@ -0,0 +1,70 @@
+import os
+import netaddr
+import logging
+
+from avocado.utils import process
+
+
+def run(test, params, env):
+    """
+    Convert remote image.
+
+    1) Start VM
+    2) Disconnect local host from the remote server one by one,
+       make sure the vm can be accessed.
+    """
+
+    def _check_hosts(hosts):
+        if len(hosts) < 2:
+            test.cancel("2 remote servers at least are required.")
+        for h in hosts:
+            if os.path.exists(h) or netaddr.valid_ipv6(h):
+                test.cancel("Neither ipv6 nor unix domain"
+                            " socket is supported by now.")
+
+    hosts = []
+    if params.get("enable_gluster") == "yes":
+        hosts.append(params["gluster_server"])
+        hosts.extend(params.get("gluster_peers", "").split())
+
+    _check_hosts(hosts)
+    hosts.pop()  # The last server should be accessible
+
+    disconn_cmd = params["disconn_cmd"]
+    recover_cmd = params["recover_cmd"]
+    conn_check_cmd = params["conn_check_cmd"]
+    disk_op_cmd = params["disk_op_cmd"]
+    disk_op_tm = int(params["disk_op_timeout"])
+
+    try:
+        session = None
+        disconn_hosts = []
+
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        tm = int(params.get("login_timeout", 300))
+        session = vm.wait_for_login(timeout=tm)
+
+        for host in hosts:
+            logging.info("Disconnect to %s" % host)
+            process.system(disconn_cmd.format(source=host),
+                           ignore_status=True, shell=True)
+            if process.system(conn_check_cmd.format(source=host),
+                              ignore_status=True, shell=True) == 0:
+                test.error("Failed to disconnect to remote server")
+            disconn_hosts.append(host)
+
+            logging.info("Do I/O in VM")
```

Comment:
Is it necessary to put the `I/O` operation into the for loop?

Reply:
yes,i/o test here is not something like a stress test, i.e. a lot of i/o operation, it just makes sure that the VM can be accessed and io can be done against the VM disk without any issue.

why putting io in a for loop? That's because we disconnect to the remote gluster server one by one in a loop, we also have to make sure the VM works well as at least one gluster server is still accessible.

Reply:
Yes, but there is a mismatch between your script and the polarion case, better to conform to the case.

Reply:
In fact, the automation is quite different with polarion case, not only for the i/o, but also for the way to make remote server unavailable, I will talk to @hellohellenmao about that. Will update the comment when we make a conclusion.

Reply:
Talked with @hellohellenmao , who confirmed that it's OK to do the test in such way, thank you.